### PR TITLE
Cleanup project wizard pt1

### PR DESCRIPTION
**Pull request by abrooksv**
* Remove unneeded custom module
* Remove hard coded sam init name
* Remove unneeded parent classes making a confusing type hierarchy

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by abrooksv** [comment]
Splitting up wizard work since it is getting rather large

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectGenerator.kt [hunk 1]
```diff
@@ -30,12 +30,14 @@ import software.aws.toolkits.resources.message
 import javax.swing.Icon
 import javax.swing.JComponent
 
-// ref: https://github.com/JetBrains/intellij-plugins/blob/master/vuejs/src/org/jetbrains/vuejs/cli/VueCliProjectGenerator.kt
+/**
+ * [ProjectTemplate] To allow for us to shim it into ProjectTemplatesFactory and use this in IntelliJ
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectGenerator.kt:34]: `DirectoryProjectGeneratorBase` so this shows up in light wizards?

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectGenerator.kt:34]: Good call

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectGenerator.kt [hunk 2]
```diff
@@ -98,16 +98,15 @@ class SamProjectGenerator :
 
     override fun createModuleBuilder(): ModuleBuilder = builder
 
-    // force the initial validation
-    override fun postponeValidation(): Boolean = false
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectGenerator.kt:102]: How did this end up not being needed?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectWizard.kt [hunk 3]
```diff
@@ -138,12 +137,10 @@ abstract class SamProjectTemplate {
         }
     }
 
-    fun getIcon() = AwsIcons.Resources.SERVERLESS_APP
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectWizard.kt:141]: what does removing this do?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectGenerator.kt (hunk 2 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectGenerator.kt:102]: Dropping of WebProjectTemplate

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectWizard.kt (hunk 3 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectWizard.kt:141]: We put an icon next to each template..it felt odd...

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectWizard.kt:141]: https://github.com/aws/aws-toolkit-jetbrains/pull/2112/files#diff-3cdf10103b9290aec22fe7703ab30973

**Review by hunterwerlla — APPROVED** [review]

**Pull request merged**